src/index.ts

```diff
@@ -0,0 +1,518 @@
+              `${context} is not a valid key context. ` +
+              `Valid keys should be of the format ${VALID_KEY_CONTEXT.toString()}.`);
+          }
+        }
+
+        if (typeof newTranslations[context] === 'object') {
+          newTranslations[context] =
+            this.cloneTranslations(<Translations>newTranslations[context]);
+        }
+      });
+
+    return newTranslations;
+  }
+
+  private deleteFrom(locale: string | string[], lookup: Rules | Translations): void {
+    const locales = this.normalizeLocales(locale);
+    const mapper = (l: string) => {
+      if (l in lookup) {
+        delete lookup[l];
+        return true;
+      } else {
+        return false;
+      }
+    };
+
+    const hasChanges = locales.map(mapper).includes(true);
+
+    if (hasChanges) {
+      this.callSubscribers();
+    }
+  }
+
+  /**
+   * Looks up the appropriate key context for the locale and number.
+   */
+  private getPluralContext(locale: string, count: number): PluralCategory {
+    const { language } = new LanguageTag(locale);
+    const rule: Rule = this.rules[locale] || this.rules[language];
+
+    // Since the locale may be determined by the end user's environment, do not throw an error. For
+    // better debugging purposes, though, print a warning stating that there is no rule for this
+    // locale.
+    if (rule === undefined) {
+      if (this.pluralWarning === undefined) {
+        this.pluralWarning = new Set();
+      }
+
+      if (!this.pluralWarning.has(language)) {
+        this.pluralWarning.add(language);
+        console.warn(`No pluralization rule found for ${locale}.`);
+      }
+    }
+
+    return rule !== undefined ? rule(count) : other();
+  }
+
+  /**
+   * Generates all possible subsets from a language tag.
+   *
+   * Examples:
+   * ```
+   * new LanguageTag('zh-Hant-HK').makeSubsets()    // ['zh-Hant', 'zh']
+   * new LanguageTag('zh-HK').makeSubsets()         // ['zh']
+   * new LanguageTag('zh-Hant').makeSubsets()       // ['zh']
+   * new LanguageTag('zh').makeSubsets()            // []
+   * ```
+   */
+  private makeSubsets(locale: string): string[] {
+    const subsets: string[] = [];
+    const languageTag = new LanguageTag(locale);
+
+    if (languageTag.hasRegion() && languageTag.hasScript()) {
+      subsets.push(`${languageTag.language}-${languageTag.script}`);
+    }
+
+    if (languageTag.language !== languageTag.toString()) {
+      subsets.push(languageTag.language);
+    }
+
+    return subsets;
+  }
+
+  private normalizeLocale(locale: string): string {
+    return new LanguageTag(locale).toString();
+  }
+
+  private normalizeLocales(locale: string | string[]): string[] {
+    if (Array.isArray(locale)) {
+      return locale.map(this.normalizeLocale);
+    } else if (!!locale) {
+      return [this.normalizeLocale(locale)];
+    } else {
+      return [];
+    }
+  }
+
+  private translationReducer(translations: Translations, context: string): TranslationLookup {
```

Comment:
You are right! Technically it could be a string. It goes through the nested `Translations` object, so I think the correct signature is `Translations | Translations[string]`.